server/src/services/poaching-incident.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable, RequestTimeoutException } from '@nestjs/common';
+import { DatabaseService } from './db.service';
+import { PoachingIncident } from '../entity/poaching-incident.entity';
+import { PoachingIncidentType } from "../entity/poaching-incident-type.entity";
+
+@Injectable()
+export class PoachingIncidentService {
+  constructor(private readonly databaseService: DatabaseService) { }
+
+  async addPoachingIncident(long: number, lat: number, pType: string): Promise<boolean> {
+    const con = await this.databaseService.getConnection();
+    const poachingIncident = new PoachingIncident();
+
+    try {
+      const poachingIncidentType = await JSON.parse(
+        JSON.stringify(await con.getRepository(PoachingIncidentType).find({ type: pType })),
+      );
+
+      poachingIncident.timestamp = new Date();
+      poachingIncident.longitude = long;
+      poachingIncident.latitude = lat;
+      poachingIncident.type = await poachingIncidentType[0]['id'];
```

Comment:
In the previous comment I showed that you can use .findOne({...}) to find only a single row.

So now you can just do this:
```
poachingIncident.type = poachingIncidentType.id;
```

Reply:
Yeah, forgot that you can find only one row